tests/e2e/contracts/subcall/subcall.go

```diff
@@ -0,0 +1,36 @@
+package subcall
+
+import (
+	_ "embed"
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	ethABI "github.com/ethereum/go-ethereum/accounts/abi"
+)
+
+// CompiledHex is the compiled subcall contract in hex encoding.
+//
+//go:embed evm_subcall_compiled.hex
+var CompiledHex string
```

Comment:
No the existing `evmtest.go` needs to be refactored because it is a total mess. I however do not want to refactor everything in this PR. Ideally each contract would have its own Go module with things exported like this and convenience wrappers to call into these contracts.